File: 启动服务器.js
#!/usr/bin/env node
/**
 * 百宝箱工具本地服务器启动脚本 (Node.js版本)
 * 运行方法: node 启动服务器.js
 * 或者: npm install -g http-server && http-server -p 8000 --cors
 */

const http = require('http');
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

// 设置端口
let PORT = 8000;

// MIME类型映射
const mimeTypes = {
    '.html': 'text/html; charset=utf-8',
    '.css': 'text/css',
    '.js': 'application/javascript',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.jpeg': 'image/jpeg',
    '.gif': 'image/gif',
    '.ico': 'image/x-icon',
    '.svg': 'image/svg+xml'
};

// 检查端口是否被占用
function isPortInUse(port) {
    return new Promise((resolve) => {
        const server = http.createServer();
        server.listen(port, () => {
            server.close(() => resolve(false));
        });
        server.on('error', () => resolve(true));
    });
}

// 寻找可用端口
async function findAvailablePort() {
    while (await isPortInUse(PORT) && PORT < 8010) {
        PORT++;
    }
    if (PORT >= 8010) {
        console.log("❌ 端口8000-8009都被占用，请手动关闭其他服务后重试");
        process.exit(1);
    }
    return PORT;
}

// 打开浏览器
function openBrowser() {
    const url = `http://localhost:${PORT}`;
    const platform = process.platform;
    
    let command;
    if (platform === 'darwin') {
        command = `open ${url}`;
    } else if (platform === 'win32') {
        command = `start ${url}`;
    } else {
        command = `xdg-open ${url}`;
    }
    
    setTimeout(() => {
        exec(command, (error) => {
            if (error) {
                console.log(`请手动在浏览器中访问: ${url}`);
            }
        });
    }, 1000);
}

// 创建服务器
async function createServer() {
    const port = await findAvailablePort();
    
    const server = http.createServer((req, res) => {
        // 设置CORS头部
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
        res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
        
        // 处理OPTIONS请求
        if (req.method === 'OPTIONS') {
            res.writeHead(200);
            res.end();
            return;
        }
        
        // 获取请求路径
        let filePath = req.url === '/' ? '/index.html' : req.url;
        filePath = path.join(__dirname, filePath);
        
        // 获取文件扩展名
        const ext = path.extname(filePath);
        const contentType = mimeTypes[ext] || 'text/plain';
        
        // 检查文件是否存在
        fs.access(filePath, fs.constants.F_OK, (err) => {
            if (err) {
                res.writeHead(404, { 'Content-Type': 'text/html; charset=utf-8' });
                res.end('404 - 文件未找到');
                return;
            }
            
            // 读取并返回文件
            fs.readFile(filePath, (err, data) => {
                if (err) {
                    res.writeHead(500, { 'Content-Type': 'text/html; charset=utf-8' });
                    res.end('500 - 服务器内部错误');
                    return;
                }
                
                res.writeHead(200, { 'Content-Type': contentType });
                res.end(data);
            });
        });
        
        // 日志记录
        console.log(`[${new Date().toISOString()}] ${req.method} ${req.url}`);
    });
    
    server.listen(port, () => {
        console.log('='.repeat(50));
        console.log('🎉 百宝箱工具服务器启动成功!');
        console.log('='.repeat(50));
        console.log(`📱 本地访问地址: http://localhost:${port}`);
        console.log(`🌐 局域网访问地址: http://你的IP地址:${port}`);
        console.log('='.repeat(50));
        console.log('💡 使用说明:');
        console.log('   1. 浏览器会自动打开网站');
        console.log('   2. 现在可以使用所有功能（翻译、OCR等）');
        console.log('   3. 按 Ctrl+C 停止服务器');
        console.log('='.repeat(50));
        
        openBrowser();
    });
    
    // 优雅关闭
    process.on('SIGINT', () => {
        console.log('\n🔴 服务器已停止');
        console.log('感谢使用百宝箱工具!');
        server.close();
        process.exit(0);
    });
}

// 检查Node.js版本
if (process.version < 'v10.0.0') {
    console.log('❌ 需要Node.js 10.0.0或更高版本');
    process.exit(1);
}

// 启动服务器
createServer().catch(error => {
    console.error('❌ 启动服务器时出错:', error.message);
    console.log('\n可能的解决方案:');
    console.log('1. 检查是否安装了Node.js');
    console.log('2. 确保端口8000未被占用');
    console.log('3. 以管理员权限运行');
    process.exit(1);
}); 

File: 启动服务器.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
百宝箱工具本地服务器启动脚本
双击运行此文件，然后在浏览器中访问 http://localhost:8000
"""

import http.server
import socketserver
import webbrowser
import os
import sys
import threading
import time

def open_browser():
    """延迟打开浏览器"""
    time.sleep(1)
    webbrowser.open('http://localhost:8000')

def main():
    # 设置端口
    PORT = 8000
    
    # 切换到脚本所在目录
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    
    # 创建HTTP请求处理器
    class MyHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
        def end_headers(self):
            # 添加CORS头部，允许跨域请求
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
            self.send_header('Access-Control-Allow-Headers', 'Content-Type')
            super().end_headers()
        
        def log_message(self, format, *args):
            # 自定义日志格式
            print(f"[{self.address_string()}] {format % args}")
    
    # 检查端口是否被占用
    def is_port_in_use(port):
        import socket
        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
            return s.connect_ex(('localhost', port)) == 0
    
    # 寻找可用端口
    while is_port_in_use(PORT) and PORT < 8010:
        PORT += 1
    
    if PORT >= 8010:
        print("❌ 端口8000-8009都被占用，请手动关闭其他服务后重试")
        input("按回车键退出...")
        return
    
    try:
        # 创建服务器
        with socketserver.TCPServer(("", PORT), MyHTTPRequestHandler) as httpd:
            print("="*50)
            print("🎉 百宝箱工具服务器启动成功!")
            print("="*50)
            print(f"📱 本地访问地址: http://localhost:{PORT}")
            print(f"🌐 局域网访问地址: http://你的IP地址:{PORT}")
            print("="*50)
            print("💡 使用说明:")
            print("   1. 浏览器会自动打开网站")
            print("   2. 现在可以使用所有功能（翻译、OCR等）")
            print("   3. 按 Ctrl+C 停止服务器")
            print("="*50)
            
            # 在新线程中打开浏览器
            browser_thread = threading.Thread(target=open_browser)
            browser_thread.daemon = True
            browser_thread.start()
            
            # 启动服务器
            httpd.serve_forever()
            
    except KeyboardInterrupt:
        print("\n")
        print("🔴 服务器已停止")
        print("感谢使用百宝箱工具!")
    except Exception as e:
        print(f"❌ 启动服务器时出错: {e}")
        print("\n可能的解决方案:")
        print("1. 检查是否安装了Python 3")
        print("2. 确保端口8000未被占用")
        print("3. 以管理员权限运行")
        input("按回车键退出...")

if __name__ == "__main__":
    main() 